### DISCUSSION TITLE
Help me understand max grid import in solar mode

### DISCUSSION THREAD

**Discussion by zulufoxtrot** [dingo35/SmartEVSE-3.5#160] [General]
Hello 👋

I have my SmartEVSE set up with these settings:
- Solar mode
- Min current: 6A
- Max grid import: 2A
- Solar start threshold: 0A

Additional info: In the winter, my solar panels provide about 6A at their best.

*What I have observed*

When charging, when the solar excess power becomes greater than the min current setting (6+A), the car draws a minimum of 8+A, which is the sum of the solar excess _and_ the max grid import. As a result, 2+A are drawn from the grid, which I don't want.

*What I'd expect*

1. When solar excess power < min current, the car draws `(solar excess + complement from the grid, as long as it's < max grid import)`
2. When solar excess power >= min current, the car draws no more than the solar excess, even if max grid import is >= 0.

Is it a bug, or a design choice? Or maybe I understood it wrong? 🤓

**Comment by dingo35** [comment]
Its a design choice.

Most people set Importcurrent 0A, StartCurrent -4A or -2A, and Stoptimer at 60 minutes.

**Comment by zulufoxtrot** [comment]
OK. Having a modest solar install, I would benefit from more granular control.

**Comment by dingo35** [comment]
You could even set StartCurrent at 0A, keep Importcurrent at 0A, Stoptimer at 60min; it would start charging as soon as your solar panels start delivering to the grid, and only goes over MinCurrent as your solar panels deliver more current.
As I understand your use case this would exactly do what you want?

**Comment by zulufoxtrot** [comment]
Thanks for the tip. I'm not sure it does exactly what I want, but I'll have to test when I get a chance.
I'll also post graphs to better explain what I want.

**Comment by zulufoxtrot** [comment]
@dingo35 it took me a while, but I managed to capture data to illustrate my issue.

Context: today was somewhat cloudy with lots of sunny spells.

Plotted data:
* solar production (which is basically identical to my solar export)
* car instant power
* smartEVSE target current (multiplied current by 230v to get matching scales)

![image](https://github.com/user-attachments/assets/dc0e0754-2528-424a-a6f5-74fcdc4a828c)

My observations:

* As long as solar excess is inferior to target current (5A in my case), the EVSE imports current from the grid to match target current. ✅ 
* As soon as solar excess equals or exceeds target power, the EVSE keeps importing current, it looks like it is importing the maximum set in ImportCurrent. 🛑 

Note: I haven't tried yet the workaround you suggested in https://github.com/dingo35/SmartEVSE-3.5/discussions/160#discussioncomment-11519042 .

**Comment by dingo35** [comment]
There is no such thing as a Target Current; there is a thing called Import Current, which is the amount that will be imported after charging has started.

I understand your way of thinking, its just not designed that way. Why dont you just try my suggestion? Its much less work than all that capturing, and probably much more effective...
